### PR TITLE
bump workflows to 24.0

### PR DESCRIPTION
**Pull request by bernt-matthias**
and use mor env variables

### CODE REVIEW

File: .github/workflows/workflow_test.yml [hunk 1]
```diff
@@ -4,14 +4,16 @@ name: Galaxy Workflow Tests for push and PR
...
+  GALAXY_BRANCH: release_24.0
 jobs:
   setup:
     name: Setup cache and determine changed repositories
     uses: ./.github/workflows/setup.yml
     with:
       python-version-list: "[\"3.11\"]"
-      galaxy-fork: galaxyproject
-      galaxy-branch: release_23.2
+      galaxy-fork: ${{ env.GALAXY_FORK }}
```
Comment (mvdbeek) [inline .github/workflows/workflow_test.yml:15]: Have you tested that this really works ? I am not sure in what order env vars and job matrices are being evaluated.

Comment (bernt-matthias) [reply .github/workflows/workflow_test.yml:15]: Nope. Is there an easy way, e.g. bump a workflow?

Comment (mvdbeek) [reply .github/workflows/workflow_test.yml:15]: `s/23.2/24.0/g`

**Comment by mvdbeek** [comment]
Attention: deployment skipped!

https://github.com/galaxyproject/iwc/actions/runs/8262108712

**Pull request merged**